### DISCUSSION TITLE
Is there a way to make `--global` dependencies written in `pyproject.toml` and `pdm.lock`?

### DISCUSSION THREAD

**Discussion by monchin** [pdm-project/pdm#2815] [Q&A]
Hello, I hope I can add `isort` and `black` globally in my projects, so I typed `pdm add -dG fmt isort -g`, but I found there is no change in `pyproject.toml` and `pdm.lock`. In fact, what I hope is that when a new user `git clone` my project, and `pdm sync`, whether `isort` and `black` is globally installed would be checked, if not, they will be installed globally automatically. I'm not sure if it is a good idea, but I think it is worth considering.

**Comment by frostming** [comment]
Not a good idea. Modifying the system libraries when installing a project isn't expected.

**Comment by huxuan** [comment]
The installation of standalone tools could be done by [pipx](http://github.com/pypa/pipx), we can use the fixed version like `pipx install isort==5.13.2 black==24.4.0` to ensure a almost same behavior (since it lacks the lock for their dependencies).

**Comment by goyalyashpal** [comment]
> whether `isort` and `black` is globally installed would be checked, if not, they will be installed globally automatically

how about "if not, they will be installed [locally, else global version will be used]"?

my use case is that,

* i use jupyter kernels for the language etc required for the project. 
* but jupyter kernels require things like `jupyter`, `ipykernel` etc which are **heavy**. 
* and installing them for each project seems like a waste. 

so, i'd like to be able to specify those to be used from global if installed globally, else install in the project locally.
or put in other words, i want to be able to install those `ipykernel` etc globally in such a way that this can be found and `import`ed by any local project

* i tried reading this, but couldn't understand: https://pdm-project.org/en/stable/usage/dependency/#manage-global-project
* it's not same as: https://github.com/pdm-project/pdm/discussions/1839
